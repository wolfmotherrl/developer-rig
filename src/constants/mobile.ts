export const MobileOrientation: { [key: string]: string } = {
  Portrait: 'Portrait',
  Landscape: 'Landscape',
}

export const DefaultMobileOrientation = MobileOrientation.Portrait;

export const MobileSizes: { [key: string]: { width: number; height: number; } } = Object.freeze({
  'iPhone X/XS Max (414x896)': {
    width: 414,
    height: 896
  },
  'iPhone X/XS (375x822)': {
    width: 375,
    height: 822
  },
  'iPhone 7/8 (375x667)': {
    width: 375,
    height: 667
  },
  'iPhone 7/8 Plus (414x736)': {
    width: 414,
    height: 736
  },
  'Google Pixel 2 (411x731)': {
    width: 411,
    height: 731
  },
  'Samsung Galaxy S8/S9 (360x740)': {
    width: 360,
    height: 740
  },
  'One Plus 3 (480x853)': {
    width: 480,
    height: 853
  },
});

export const DefaultMobileSize = 'iPhone X/XS (375x822)';

