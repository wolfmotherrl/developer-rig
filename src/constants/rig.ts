export const RigRole = 'rig';
export const DeveloperRigUserId = '265737932';

export enum LocalStorageKeys {
  CurrentProjectIndex = 'currentProjectIndex',
  ExtensionViews = 'extensionViews',
  Projects = 'projects',
  RigLogin = 'rigLogin',
}
