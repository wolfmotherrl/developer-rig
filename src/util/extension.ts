import { createSignedToken, TokenSpec } from './token';
import { ExtensionManifest } from '../core/models/manifest';
import { generateId } from './id';
import { RigRole } from '../constants/rig';
import { ViewerTypes } from '../constants/viewer-types';
import { fetchExtensionManifest } from './api';

export async function fetchUserExtensionManifest(isLocal: boolean, userId: string, secret: string, clientId: string, version: string): Promise<ExtensionManifest> {
  const tokenSpec: TokenSpec = {
    role: RigRole,
    secret,
    userId,
  };
  const token = createSignedToken(tokenSpec);
  return await fetchExtensionManifest(isLocal, clientId, version, token);
}

export function createExtensionToken(newRole: string, linkedUserId: string, channelId: string, secret: string, opaqueId: string): string {
  const opaque = opaqueId ? opaqueId : generateId(15);
  switch (newRole) {
    case ViewerTypes.LoggedOut:
      return createSignedToken({ role: 'viewer', opaqueUserId: 'ARIG' + opaque, channelId, secret });
    case ViewerTypes.LoggedIn:
      return createSignedToken({ role: 'viewer', opaqueUserId: 'URIG' + opaque, userId: linkedUserId, channelId, secret });
    case ViewerTypes.Broadcaster:
      return createSignedToken({ role: 'broadcaster', opaqueUserId: 'URIG' + opaque, userId: channelId, channelId, secret });
    default:
      return createSignedToken({ role: RigRole, opaqueUserId: 'ARIG' + opaque, channelId, secret });
  }
}

export function createExtensionObject(
  manifest: ExtensionManifest,
  index: string,
  role: string,
  linkedUserId: string,
  channelId: string,
  secret: string,
  opaqueId: string
): ExtensionCoordinator.ExtensionObject {
  return {
    authorName: manifest.authorName,
    bitsEnabled: manifest.bitsEnabled,
    clientId: manifest.id,
    description: manifest.description,
    hasChatSupport: false,
    iconUrl: manifest.iconUrls['100x100'],
    iconUrls: { square100: manifest.iconUrls['100x100'] },
    id: manifest.id + ':' + index,
    name: manifest.name,
    requestIdentityLink: manifest.requestIdentityLink,
    sku: manifest.sku,
    state: manifest.state as ExtensionCoordinator.ExtensionState,
    summary: manifest.summary,
    token: createExtensionToken(role, linkedUserId, channelId, secret, opaqueId),
    vendorCode: manifest.vendorCode,
    version: manifest.version,
    views: manifest.views,
    whitelistedConfigUrls: manifest.whitelistedConfigUrls,
    whitelistedPanelUrls: manifest.whitelistedPanelUrls,
  };
}
