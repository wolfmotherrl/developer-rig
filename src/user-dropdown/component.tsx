import * as React from 'react';
import * as classNames from 'classnames';
import { UserSession } from '../core/models/user-session';
import { fetchNewRelease } from '../util/api';
import * as reddot from '../img/reddot.svg';
import * as whitetriangle from '../img/whitetriangle.svg';
import './component.sass';
import { LocalStorageKeys } from '../constants/rig';

export interface PublicProps {
  session: UserSession;
}

export interface ReduxDispatchProps {
  logout: () => void;
}

interface State {
  open: boolean;
  showingNewRelease: boolean;
  releaseUrl?: string;
}

type Props = PublicProps & ReduxDispatchProps;
export class UserDropdownComponent extends React.Component<Props, State> {
  public state: State = {
    open: false,
    showingNewRelease: false,
  }

  private signOut = () => {
    localStorage.removeItem(LocalStorageKeys.RigLogin);
    this.props.logout();
  }

  private toggleDropdown = () => {
    this.setState({
      open: !this.state.open,
    })
  }

  public componentDidMount() {
    if (process.env.GIT_RELEASE) {
      fetchNewRelease()
        .then((result) => {
          if (result.tagName !== process.env.GIT_RELEASE) {
            this.setState({
              showingNewRelease: true,
              releaseUrl: result.zipUrl,
            });
          }
        });
    }
  }

  public render() {
    if (!this.props.session) {
      return null;
    }
    const { login, profileImageUrl } = this.props.session;
    const usernameClasses = classNames({
      'user-dropdown__username': true,
      'open': this.state.open,
    });
    const dropdownClass = classNames({
      'user-dropdown__menu': true,
      'transition': true,
      'open': this.state.open,
    });
    const triangleClass = classNames({
      'user-dropdown__triangle': true,
      'open': this.state.open,
    });

    return (
      <div onClick={this.toggleDropdown} className='user-dropdown' tabIndex={0}>
        <div className='user-dropdown__name-container'>
          {this.state.showingNewRelease && (
            <img alt='!' title='Rig Update Available' src={reddot} width='8' height='8' />
          )}
          <img alt={login} className='user-dropdown__image' src={profileImageUrl} />
          <div className={usernameClasses}>{login}</div>
          <img src={whitetriangle} className={triangleClass} />
        </div>
        <div className={dropdownClass}>
          <ul>
            <li>
              <a target='_blank' href="https://dev.twitch.tv/dashboard/extensions">Extensions Dashboard</a>
            </li>
            <li>
              <a target='_blank' href="https://dev.twitch.tv/docs/extensions/">Documentation</a>
            </li>
            <li>
              <a href='mailto:developers@twitch.tv'>Give Feedback</a>
            </li>
            <li>
              <a target='_blank' href={this.state.releaseUrl}>
                {this.state.showingNewRelease ? 'Rig Update Available' : 'Rig Up To Date'}
                {this.state.showingNewRelease && (
                  <img alt='!' src={reddot} width='8' height='8' />
                )}
              </a>
            </li>
            <li><div className='user-dropdown__divider'></div></li>
            <li onClick={() => { this.signOut() }}>Sign Out</li>
          </ul>
        </div>
      </div>
    );
  }
}
