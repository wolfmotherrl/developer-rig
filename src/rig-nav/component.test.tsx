import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigNavComponent } from '.';
import { NavItem } from '../constants/nav-items';
import { LoginButton } from '../login-button';
import { RigProject } from '../core/models/rig';
import { UserDropdown } from '../user-dropdown';
import { createExtensionManifestForTest } from '../tests/constants/extension';

const defaultGenerator = () => ({
  openConfigurationsHandler: jest.fn(),
  viewerHandler: jest.fn(),
  configHandler: jest.fn(),
  liveConfigHandler: jest.fn(),
  openProductManagementHandler: jest.fn(),
  selectedView: NavItem.ExtensionViews,
  error: '',
  manifest: createExtensionManifestForTest(),
  session: { displayName: 'test', login: 'test', id: 'test', profileImageUrl: 'test.png', authToken: 'test' },
  mockApiEnabled: false,
  currentProjectIndex: 0,
  projects: [] as RigProject[],
  createNewProject: () => {},
  deleteProject: jest.fn(),
  selectProject: (_projectIndex: number) => {},
});


const setupShallow = setupShallowTest(RigNavComponent, defaultGenerator);

describe('<RigNavComponent />', () => {
  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an error', () => {
    const { wrapper } = setupShallow({
      error: 'test error',
    });

    expect(wrapper.find('.top-nav-error').text().trim()).toBe('test error');
  });

  it('invokes deleteProject', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.personal-bar__button').first().simulate('click');
    expect(wrapper.instance().props.deleteProject).toHaveBeenCalled();
  });

  it('correctly handles clicks on each tab', () => {
    const { wrapper } = setupShallow();
    wrapper.find('a.top-nav-item').forEach((tab: any) => {
      tab.simulate('click');
    });
    expect(wrapper.instance().props.viewerHandler).toHaveBeenCalledTimes(3);
  });

  it('correct css classes are set when things are selected', () => {
    const { wrapper } = setupShallow({
      selectedView: NavItem.ExtensionViews,
    });
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);

    wrapper.setProps({
      selectedView: NavItem.ProductManagement,
    });
    wrapper.update();
    expect(wrapper.find('.top-nav-item__selected')).toHaveLength(1);
  });

  it('renders login button if no session', () => {
    const { wrapper } = setupShallow({
      session: undefined,
    });
    expect(wrapper.find(LoginButton));
  });

  it('renders user dropdown if session exists', () => {
    const { wrapper } = setupShallow();
    expect(wrapper.find(UserDropdown));
  });

  it('disables product management tab when user is not logged in', () => {
    const { wrapper } = setupShallow({
      session: undefined,
    });
    expect(wrapper.find('.top-nav-item__disabled')).toHaveLength(2);
  });

  it('disables product management tab when extension is not bits enabled', () => {
    const { wrapper } = setupShallow({
      manifest: {
        ...createExtensionManifestForTest(),
        bitsEnabled: false,
      },
    });
    expect(wrapper.find('.top-nav-item__disabled')).toHaveLength(2);
  });
});
