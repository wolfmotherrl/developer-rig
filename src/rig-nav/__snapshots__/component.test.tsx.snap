// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigNavComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigNavComponent
    configHandler={[Function]}
    createNewProject={[Function]}
    currentProjectIndex={0}
    deleteProject={[Function]}
    error=""
    liveConfigHandler={[Function]}
    manifest={
        Object {
            "authorName": "test",
            "bitsEnabled": true,
            "configurationLocation": "custom",
            "description": "test",
            "hasChatSupport": false,
            "iconUrls": Object {
              "100x100": "test",
            },
            "id": "test",
            "name": "test",
            "requestIdentityLink": false,
            "sku": "test",
            "state": "test",
            "summary": "test",
            "vendorCode": "test",
            "version": "0.0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "size": 1024,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 24,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [],
            "whitelistedPanelUrls": Array [],
          }
    }
    mockApiEnabled={false}
    openConfigurationsHandler={[Function]}
    openProductManagementHandler={[Function]}
    projects={Array []}
    selectProject={[Function]}
    selectedView={0}
    session={
        Object {
            "authToken": "test",
            "displayName": "test",
            "id": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    viewerHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="personal-bar"
>
          <ProjectDropdown
                    createNewProject={[Function]}
                    currentProjectIndex={0}
                    projects={Array []}
                    selectProject={[Function]}
          />
          <div
                    className="personal-bar__ext-name"
          >
                    <span>
                              test
                    </span>
                    <button
                              className="personal-bar__button"
                              onClick={[Function]}
                    >
                              Delete
                    </button>
          </div>
          <div
                    className="top-nav-item__login"
          >
                    <Connect(UserDropdownComponent)
                              session={
                                        Object {
                                                  "authToken": "test",
                                                  "displayName": "test",
                                                  "id": "test",
                                                  "login": "test",
                                                  "profileImageUrl": "test.png",
                                                }
                              }
                    />
          </div>
</div>,
        <div
          className="top-nav__item-container"
>
          <a
                    className="offset top-nav-item"
                    onClick={[Function]}
          >
                    Project Overview
          </a>
          <a
                    className="offset top-nav-item top-nav-item__selected"
                    onClick={[Function]}
          >
                    Extension Views
          </a>
          <a
                    className="offset top-nav-item"
                    onClick={[Function]}
          >
                    Manage Bits Products
          </a>
          <a
                    className="offset top-nav-item top-nav-item__disabled"
                    onClick={[Function]}
          >
                    Configuration Service
          </a>
</div>,
      ],
      "className": "top-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ProjectDropdown
              createNewProject={[Function]}
              currentProjectIndex={0}
              projects={Array []}
              selectProject={[Function]}
/>,
            <div
              className="personal-bar__ext-name"
>
              <span>
                            test
              </span>
              <button
                            className="personal-bar__button"
                            onClick={[Function]}
              >
                            Delete
              </button>
</div>,
            <div
              className="top-nav-item__login"
>
              <Connect(UserDropdownComponent)
                            session={
                                          Object {
                                                        "authToken": "test",
                                                        "displayName": "test",
                                                        "id": "test",
                                                        "login": "test",
                                                        "profileImageUrl": "test.png",
                                                      }
                            }
              />
</div>,
          ],
          "className": "personal-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "createNewProject": [Function],
              "currentProjectIndex": 0,
              "projects": Array [],
              "selectProject": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  test
</span>,
                <button
                  className="personal-bar__button"
                  onClick={[Function]}
>
                  Delete
</button>,
              ],
              "className": "personal-bar__ext-name",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "className": "personal-bar__button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(UserDropdownComponent)
                session={
                                Object {
                                                "authToken": "test",
                                                "displayName": "test",
                                                "id": "test",
                                                "login": "test",
                                                "profileImageUrl": "test.png",
                                              }
                }
/>,
              "className": "top-nav-item__login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "session": Object {
                  "authToken": "test",
                  "displayName": "test",
                  "id": "test",
                  "login": "test",
                  "profileImageUrl": "test.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="offset top-nav-item"
              onClick={[Function]}
>
              Project Overview
</a>,
            <a
              className="offset top-nav-item top-nav-item__selected"
              onClick={[Function]}
>
              Extension Views
</a>,
            <a
              className="offset top-nav-item"
              onClick={[Function]}
>
              Manage Bits Products
</a>,
            <a
              className="offset top-nav-item top-nav-item__disabled"
              onClick={[Function]}
>
              Configuration Service
</a>,
          ],
          "className": "top-nav__item-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Project Overview",
              "className": "offset top-nav-item",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Project Overview",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Extension Views",
              "className": "offset top-nav-item top-nav-item__selected",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Extension Views",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Manage Bits Products",
              "className": "offset top-nav-item",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Manage Bits Products",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Configuration Service",
              "className": "offset top-nav-item top-nav-item__disabled",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Configuration Service",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="personal-bar"
>
            <ProjectDropdown
                        createNewProject={[Function]}
                        currentProjectIndex={0}
                        projects={Array []}
                        selectProject={[Function]}
            />
            <div
                        className="personal-bar__ext-name"
            >
                        <span>
                                    test
                        </span>
                        <button
                                    className="personal-bar__button"
                                    onClick={[Function]}
                        >
                                    Delete
                        </button>
            </div>
            <div
                        className="top-nav-item__login"
            >
                        <Connect(UserDropdownComponent)
                                    session={
                                                Object {
                                                            "authToken": "test",
                                                            "displayName": "test",
                                                            "id": "test",
                                                            "login": "test",
                                                            "profileImageUrl": "test.png",
                                                          }
                                    }
                        />
            </div>
</div>,
          <div
            className="top-nav__item-container"
>
            <a
                        className="offset top-nav-item"
                        onClick={[Function]}
            >
                        Project Overview
            </a>
            <a
                        className="offset top-nav-item top-nav-item__selected"
                        onClick={[Function]}
            >
                        Extension Views
            </a>
            <a
                        className="offset top-nav-item"
                        onClick={[Function]}
            >
                        Manage Bits Products
            </a>
            <a
                        className="offset top-nav-item top-nav-item__disabled"
                        onClick={[Function]}
            >
                        Configuration Service
            </a>
</div>,
        ],
        "className": "top-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ProjectDropdown
                createNewProject={[Function]}
                currentProjectIndex={0}
                projects={Array []}
                selectProject={[Function]}
/>,
              <div
                className="personal-bar__ext-name"
>
                <span>
                                test
                </span>
                <button
                                className="personal-bar__button"
                                onClick={[Function]}
                >
                                Delete
                </button>
</div>,
              <div
                className="top-nav-item__login"
>
                <Connect(UserDropdownComponent)
                                session={
                                                Object {
                                                                "authToken": "test",
                                                                "displayName": "test",
                                                                "id": "test",
                                                                "login": "test",
                                                                "profileImageUrl": "test.png",
                                                              }
                                }
                />
</div>,
            ],
            "className": "personal-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "createNewProject": [Function],
                "currentProjectIndex": 0,
                "projects": Array [],
                "selectProject": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    test
</span>,
                  <button
                    className="personal-bar__button"
                    onClick={[Function]}
>
                    Delete
</button>,
                ],
                "className": "personal-bar__ext-name",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test",
                  },
                  "ref": null,
                  "rendered": "test",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "className": "personal-bar__button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(UserDropdownComponent)
                  session={
                                    Object {
                                                      "authToken": "test",
                                                      "displayName": "test",
                                                      "id": "test",
                                                      "login": "test",
                                                      "profileImageUrl": "test.png",
                                                    }
                  }
/>,
                "className": "top-nav-item__login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "session": Object {
                    "authToken": "test",
                    "displayName": "test",
                    "id": "test",
                    "login": "test",
                    "profileImageUrl": "test.png",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="offset top-nav-item"
                onClick={[Function]}
>
                Project Overview
</a>,
              <a
                className="offset top-nav-item top-nav-item__selected"
                onClick={[Function]}
>
                Extension Views
</a>,
              <a
                className="offset top-nav-item"
                onClick={[Function]}
>
                Manage Bits Products
</a>,
              <a
                className="offset top-nav-item top-nav-item__disabled"
                onClick={[Function]}
>
                Configuration Service
</a>,
            ],
            "className": "top-nav__item-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Project Overview",
                "className": "offset top-nav-item",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Project Overview",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Extension Views",
                "className": "offset top-nav-item top-nav-item__selected",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Extension Views",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Manage Bits Products",
                "className": "offset top-nav-item",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Manage Bits Products",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Configuration Service",
                "className": "offset top-nav-item top-nav-item__disabled",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Configuration Service",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
